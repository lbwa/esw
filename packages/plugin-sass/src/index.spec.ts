/* eslint-disable jest/expect-expect */
import fs from 'fs/promises'
import path from 'path'
import { build } from 'esbuild'
import type esbuild from 'esbuild'
import sassPlugin from '.'

const cacheDir = path.resolve(__dirname, './.cache')

async function run(name: string, options: esbuild.BuildOptions = {}) {
  const outDir = `${cacheDir}/${name}`
  const result = await build({
    entryPoints: ['./fixtures/index.sass'],
    outdir: outDir,
    plugins: [sassPlugin()],
    absWorkingDir: __dirname,
    ...options
  })
  expect(result.errors).toHaveLength(0)
  expect(result.warnings).toHaveLength(0)
  const output = await fs.readFile(
    path.resolve(__dirname, outDir, `./index.css`),
    { encoding: 'utf8' }
  )
  return {
    result,
    output
  }
}

const getTestName = () =>
  expect
    .getState()
    .currentTestName.replace('sass plugin ', '')
    .replace(/\s/g, '-')

beforeAll(async () => {
  await fs.rm(cacheDir, { force: true, recursive: true })
})

describe('sass plugin', () => {
  it('should work without bundle', async () => {
    const { output } = await run(getTestName())
    expect(output).toMatchSnapshot(getTestName())
  })

  it('should work with bundle', async () => {
    const { output } = await run(getTestName(), { bundle: true })
    expect(output).toContain(`
html,
body {
  font-size: 16px;
}
`)
    expect(output).toContain(`
html,
body {
  color: var(--theme-primary);
}
`)
  })

  it('should work with css minify and bundle', async () => {
    const { output } = await run(getTestName(), { bundle: true, minify: true })
    expect(output).toMatchSnapshot(getTestName())
  })
})
