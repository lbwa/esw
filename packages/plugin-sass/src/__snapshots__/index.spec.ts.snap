// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`sass plugin should work with css minify and bundle: should-work-with-css-minify-and-bundle 1`] = `
"html,body{font-size:16px}html,body{color:var(--theme-primary)}
"
`;

exports[`sass plugin should work without bundle: should-work-without-bundle 1`] = `
"@import \\"./dep.css\\";
html,
body {
  color: var(--theme-primary);
}
"
`;
